python/pyiceberg/table/__init__.py

```diff
@@ -167,6 +169,32 @@ def __eq__(self, other: Any) -> bool:
         )
 
 
+class StaticTable(Table):
+    """Load a table directly from a metadata file (i.e., without using a catalog)."""
+
+    def refresh(self) -> Table:
+        """StaticTable metadata cannot be refreshed."""
+        raise StaticTableImmutableError("StaticTable metadata cannot be refreshed.")
+
+    @classmethod
+    def from_metadata(cls, metadata_location: str, properties: Properties = EMPTY_DICT) -> StaticTable:
+
+        metadata = cls._load_metadata(metadata_location, properties)
+
+        return cls(
+            identifier=("static-table", metadata_location),
+            metadata_location=metadata_location,
+            metadata=metadata,
+            io=load_file_io({**properties, **metadata.properties}),
+        )
+
+    @staticmethod
+    def _load_metadata(metadata_location: str, properties: Properties) -> TableMetadata:
```

Comment:
How do you feel about merging this logic into `from_metadata`? I t looks like we don't reuse this anywhere.